pkg/admissionreview/handler.go

```diff
@@ -0,0 +1,123 @@
+package admissionreview
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	codecs serializer.CodecFactory
+)
+
+func init() {
+	utilruntime.Must(admissionv1.AddToScheme(scheme))
+
+	codecs = serializer.NewCodecFactory(scheme)
+}
+
+type HandleFunc func(*v1.AdmissionReview) error
+
+type handler struct {
```

Comment:
nit: you can implement methods directly on HandleFunc without having a need of additional struct

Reply:
that would work as well. this layout comes from supporting multiple api versions so you could call different handles for different api groups